homeassistant/components/cover/somfy.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for Somfy Covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.somfy/
+"""
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION, \
+    ATTR_TILT_POSITION
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES
+
+DEPENDENCIES = ['somfy']
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Somfy cover platform."""
+    from pymfy.api.devices.category import Category
+    categories = {Category.ROLLER_SHUTTER.value, Category.INTERIOR_BLIND.value,
+                  Category.EXTERIOR_BLIND.value}
+
+    devices = hass.data[DOMAIN][DEVICES]
+    for cover in devices:
+        if categories & set(cover.categories):
+            add_entities([SomfyCover(cover, hass)])
+
+
+class SomfyCover(SomfyEntity, CoverDevice):
+    """Representation of a Somfy cover device."""
+
+    def close_cover(self, **kwargs):
+        """Close the cover."""
+        from pymfy.api.devices.roller_shutter import RollerShutter
+        RollerShutter(self.device, self.api).close()
```

Comment:
Why do you keep instantiating the RollerShutter object ? Why not keep him around ?